### PR TITLE
Nisq analyzer plugin

### PR DESCRIPTION
**Pull request by cloudsftp**
Implementation of communication with QHana

### CODE REVIEW

File: src/app/components/implementation-view/qpu-selection/qpu-selection.component.ts [hunk 1]
```diff
@@ -225,7 +227,11 @@ export class QpuSelectionComponent implements OnInit, AfterViewInit {
   }
 
   onAddAnalysis(): void {
-    this.refreshNisqImpl();
+    if (this.impl) { // TODO: Properly check wheather this.impl comes from qhana plugin
```
Comment (salmma) [inline src/app/components/implementation-view/qpu-selection/qpu-selection.component.ts]: solve TODO

File: src/app/components/navigation/navigation.component.html [hunk 2]
```diff
@@ -1,22 +1,24 @@
 <div>
     <mat-toolbar color="primary" id="pa_header">
-        <a href="https://platform.planqk.de/my-content">
+        <a *ngIf="!plugin.isPlugin" href="https://platform.planqk.de/my-content">
```
Comment (salmma) [inline src/app/components/navigation/navigation.component.html]: besides hiding the PlanQK button and the logo, please also change the color of the top bar to one suitable color, as also this shaded bar color is from PlanQK

**Review by salmma — CHANGES_REQUESTED** [review]
see comments

File: src/app/components/services/qhana-plugin.service.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  /**
+   * Measure the current height and inform the parent window if the height has changed.
+   */
+  notifyParentWindowOnHeightChange(): void {
+    if (this.isPlugin) {
+      const height = Math.max(
+        document.body.scrollHeight,
+        document.documentElement.scrollHeight
+      );
+      if (height !== this.qhanaFrontendState.height) {
```
Comment (buehlefs) [inline src/app/components/services/qhana-plugin.service.ts]: This if is only relevant if you update the height in the frontend state

File: src/app/components/services/qhana-plugin.service.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        this.sendMessage({ type: 'ui-resize', height });
+      }
+    }
+  }
+
+  initializePlugin(): void {
+    // prevent double execution if script is already loaded in the current window
+    if (!this.qhanaFrontendState.initialized) {
```
Comment (buehlefs) [inline src/app/components/services/qhana-plugin.service.ts:160]: This does not prevent double execution without the variable being changed somewhere!

File: src/app/components/services/qhana-plugin.service.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    data.urls.forEach((url) => {
+      const styleLink = document.createElement('link');
+      styleLink.href = url;
+      styleLink.rel = 'stylesheet';
+      head.appendChild(styleLink);
+    });
+    document.body.style.background = 'transparent';
+  }
+  
+  getUUID(impl: ImplementationItem, uuids: Map<string, string>): string {
```
Comment (buehlefs) [inline src/app/components/services/qhana-plugin.service.ts]: please use visibilities like private to hide internal methods. This helps to reduce the API surface and makes it easier to choose the correct method. Also document all public (and also all private) methods.

**Review by buehlefs — CHANGES_REQUESTED** [review]

File: src/app/components/navigation/navigation.component.html [hunk 6]
```diff
@@ -1,13 +1,13 @@
 <div>
-    <mat-toolbar color="primary" id="pa_header">
-        <a *ngIf="!plugin.isPlugin" href="https://platform.planqk.de/my-content">
+    <mat-toolbar color="primary" [id]="qhanaService.isPlugin ? 'pa_header' : 'pa_header_pqk'">
```
Comment (buehlefs) [inline src/app/components/navigation/navigation.component.html]: Ids should not be changed to achieve styling differences. It is always better to use css classes if styles should change!

**Review by buehlefs — APPROVED** [review]

File: src/app/components/navigation/navigation.component.scss [hunk 7]
```diff
@@ -1,9 +1,6 @@
-#pa_header_pqk {
+.pa_header {
```
Comment (buehlefs) [inline src/app/components/navigation/navigation.component.scss]: prefer `-` over `_` in css class names
If you change this, you can merge without requesting another review

**Review by buehlefs — APPROVED** [review]

**Pull request merged**